homeassistant/components/bond/utils.py

```diff
@@ -89,31 +91,40 @@ class BondHub:
     def __init__(self, bond: Bond):
         """Initialize Bond Hub."""
         self.bond: Bond = bond
+        self._bridge: Optional[dict] = None
         self._version: Optional[dict] = None
         self._devices: Optional[List[BondDevice]] = None
 
-    async def setup(self):
+    async def setup(self, max_devices=None):
         """Read hub version information."""
         self._version = await self.bond.version()
         _LOGGER.debug("Bond reported the following version info: %s", self._version)
-
         # Fetch all available devices using Bond API.
         device_ids = await self.bond.devices()
-        self._devices = [
-            BondDevice(
-                device_id,
-                await self.bond.device(device_id),
-                await self.bond.device_properties(device_id),
+        self._devices = []
+        for idx, device_id in enumerate(device_ids):
+            if max_devices is not None and idx >= max_devices:
+                break
+
+            device, props = await asyncio.gather(
+                self.bond.device(device_id), self.bond.device_properties(device_id)
             )
-            for device_id in device_ids
-        ]
+
+            self._devices.append(BondDevice(device_id, device, props))
 
         _LOGGER.debug("Discovered Bond devices: %s", self._devices)
+        try:
+            # Smart by bond devices do not have a bridge api call
+            self._bridge = await self.bond.bridge()
+        except ClientResponseError:
+            self._bridge = {}
+        _LOGGER.debug("Bond reported the following bridge info: %s", self._bridge)
 
     @property
     def bond_id(self) -> str:
```

Comment:
```suggestion
    def bond_id(self) -> Optional[str]:
```

Does this mean that the name for config entry and devices need another fallback?

Reply:
Nope, we abort if there is no bond id and tell them the firmware is too old